docs/design/ROADMAP.md

```
@@ -1932,6 +1932,295 @@ CockroachDB's BIGSERIAL emulation works correctly with our PostgreSQL migrations
+
+**Duration Estimate**: 9 weeks
+
+**Status**: 📋 Planned
+
+**Design Document**: [NATS_ASYNC_WRITES_DESIGN.md](NATS_ASYNC_WRITES_DESIGN.md)
+
+### Overview
+
+NATS-first write architecture where writes go to NATS JetStream first, then are consumed and batched into storage:
+
+```
```

markdownlint-cli2 (0.20.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)